src/plugins/data/public/ui/query_editor/query_editor.tsx

```diff
@@ -371,9 +352,11 @@ export default class QueryEditorUI extends Component<Props, State> {
       provideCompletionItems: this.provideCompletionItems,
     };
 
+    const languageEditorFunc = this.languageManager.getLanguage(this.props.query.language)!.editor;
+
     const languageEditor = useQueryEditor
-      ? createDefaultEditor(singleLineInputProps, {}, defaultInputProps)
-      : createDQLEditor(singleLineInputProps, singleLineInputProps, {
+      ? languageEditorFunc(singleLineInputProps, {}, defaultInputProps)
```

Comment:
like above i think this should already being the editor wired up with state but will need to export it properly from the UI service.

can be a clean up follow up non blocker